WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewModel.swift

```diff
@@ -20,29 +71,99 @@ class InPersonPaymentsMenuViewModel: ObservableObject {
         return ResultsController<StoragePaymentGateway>(storageManager: ServiceLocator.storageManager,
                                                         matching: predicate,
                                                         sortedBy: [])
-    }()
+    }
 
-    init(stores: StoresManager = ServiceLocator.stores) {
-        self.stores = stores
-        observePaymentGateways()
+    private func updateCashOnDeliveryEnabledState() {
+        cashOnDeliveryEnabledState = cashOnDeliveryGateway?.enabled ?? false
     }
 
-    private func observePaymentGateways() {
-        guard let paymentGatewaysFetchedResultsController = paymentGatewaysFetchedResultsController else {
+    private var cashOnDeliveryGateway: PaymentGateway? {
+        paymentGatewaysFetchedResultsController?.fetchedObjects.first(where: {
+            $0.gatewayID == PaymentGateway.Constants.cashOnDeliveryGatewayID
+        })
+    }
+
+    // MARK: - Toggle Cash on Delivery Payment Gateway
+
+    func updateCashOnDeliverySetting(enabled: Bool) {
+        switch enabled {
+        case true:
+            enableCashOnDeliveryGateway()
+        case false:
+            disableCashOnDeliveryGateway()
+        }
+    }
+
+    private func enableCashOnDeliveryGateway() {
```

Comment:
Not a blocker, I see that `enableCashOnDeliveryGateway` and `disableCashOnDeliveryGateway` implementations are very similar, isn't there a way to collapse them just in one method passing a boolean parameter?

Reply:
I'll have another look... I avoided doing that to make it a bit clearer that they do different things (enable creates a new PaymentGateway to overwrite the existing, disable takes the existing gateway and changes one property)...

I'm not sure it's that much clearer though.